Index: backports-2017-11-01/net/Makefile
===================================================================
--- backports-2017-11-01.orig/net/Makefile
+++ backports-2017-11-01/net/Makefile
@@ -10,6 +10,9 @@ obj-$(CONFIG_NET)		:= socket.o core/
 tmp-$(CONFIG_COMPAT) 		:= compat.o
 obj-$(CONFIG_NET)		+= $(tmp-y)
 
+# added by dtc ------
+obj-y   += ../../../../../common_folder/dtc_debugfs/
+
 # LLC has to be linked before the files in net/802/
 obj-$(CONFIG_LLC)		+= llc/
 obj-$(CONFIG_NET)		+= ethernet/ 802/ sched/ netlink/ bpf/
Index: backports-2017-11-01/net/mac80211/Makefile
===================================================================
--- backports-2017-11-01.orig/net/mac80211/Makefile
+++ backports-2017-11-01/net/mac80211/Makefile
@@ -29,7 +29,8 @@ mac80211-y := \
 	chan.o \
 	trace.o mlme.o \
 	tdls.o \
-	ocb.o
+	ocb.o \
+    ../../../../../../common_folder/dtc_debugfs/dtc_debugfs.o
 
 mac80211-$(CPTCFG_MAC80211_LEDS) += led.o
 mac80211-$(CPTCFG_MAC80211_DEBUGFS) += \
Index: backports-2017-11-01/net/mac80211/main.c
===================================================================
--- backports-2017-11-01.orig/net/mac80211/main.c
+++ backports-2017-11-01/net/mac80211/main.c
@@ -33,6 +33,9 @@
 #include "led.h"
 #include "debugfs.h"
 
+// added by dtc ------
+#include "../../../../../../common_folder/dtc_debugfs/dtc_debugfs.h"
+
 void ieee80211_configure_filter(struct ieee80211_local *local)
 {
 	u64 mc;
@@ -1263,6 +1266,13 @@ static int __init ieee80211_init(void)
 	if (ret)
 		goto err_netdev;
 
+    // added by dtc ------
+    if (dtc_init_debugfs("dtcMac", 2) == 0) {
+        printk(KERN_ALERT "dtc: in ieee80211_init, init_debugfs succeeds\n");
+    } else {
+        printk(KERN_ALERT "dtc: in ieee80211_init, init_debugfs fails\n");
+    }
+
 	return 0;
  err_netdev:
 	rc80211_minstrel_exit();
@@ -1272,6 +1282,9 @@ static int __init ieee80211_init(void)
 
 static void __exit ieee80211_exit(void)
 {
+    // added by dtc ------
+    dtc_cleanup_debugfs();
+
 	rc80211_minstrel_exit();
 
 	ieee80211s_stop();
Index: backports-2017-11-01/net/mac80211/tx.c
===================================================================
--- backports-2017-11-01.orig/net/mac80211/tx.c
+++ backports-2017-11-01/net/mac80211/tx.c
@@ -39,6 +39,9 @@
 #include "wme.h"
 #include "rate.h"
 
+// added by dtc
+#include "../../../../../../common_folder/dtc_debugfs/dtc_debugfs.h"
+
 /* misc utils */
 
 static inline void ieee80211_tx_stats(struct net_device *dev, u32 len)
@@ -3766,6 +3769,15 @@ netdev_tx_t ieee80211_subif_start_xmit(s
 		skb->sk->sk_pacing_shift = 6;
 #endif
 
+    // added by dtc ------
+    if (dtc_debugfs_enable & DTC_ENABLE_UDP &&
+            dtc_debugfs_time_loc & DTC_TIME_MAC_SEND &&
+            ((struct iphdr*) skb_network_header(skb))->protocol == IPPROTO_UDP &&
+            dtc_debugfs_target_ip == ((struct iphdr*) skb_network_header(skb))->daddr &&
+            dtc_debugfs_target_port == ((struct udphdr*) skb_transport_header(skb))->dest) {
+        dtc_debugfs_log1((u8*) dtc_debugfs_skb_udp_payload(skb), 4);        
+    }
+
 	if (unlikely(ieee80211_multicast_to_unicast(skb, dev))) {
 		struct sk_buff_head queue;
 
